crates/solidity/src/build/contract.rs

```diff
@@ -139,6 +139,9 @@ impl Contract {
         let bytecode = hex::encode(self.build.bytecode.as_slice());
         if let Some(evm) = standard_json_contract.evm.as_mut() {
             evm.modify(assembly_text, bytecode);
+            if standard_json_contract.ir_optimized.is_some() {
+                evm.assembly = None;
```

Comment:
@xermicus I am not sure this, can I assume if ir_optimized is set, the legacyassembly as ir is not needed?

Reply:
ir-optimized is just another output option. They are two different things.

Reply:
OK, removed